### PR TITLE
Modified turn code to make sure on error condition reward is distributed equally

### PR DESCRIPTION
**Pull request by bovard**
When an agent times out or has a bad response, make sure that agent loses and all others have the same reward.

https://b.corp.google.com/issues/343793215

### CODE REVIEW

**Comment by bovard** [comment]
```
============================= test session starts ==============================
platform linux -- Python 3.10.13, pytest-8.2.1, pluggy-1.5.0
rootdir: /usr/src/app/kaggle_environments
plugins: typeguard-4.1.5, anyio-4.2.0
collected 80 items

kaggle_environments/envs/connectx/test_connectx.py ..............        [ 17%]
kaggle_environments/envs/halite/test_halite.py ..............            [ 35%]
kaggle_environments/envs/kore_fleets/test_kore_fleets.py ............... [ 53%]
.....                                                                    [ 60%]
kaggle_environments/envs/llm_20_questions/test_llm_20_questions.py ....  [ 65%]
kaggle_environments/envs/lux_ai_2021/test_lux.py ..                      [ 67%]
kaggle_environments/envs/lux_ai_s2/test_lux.py .                         [ 68%]
kaggle_environments/envs/rps/test_rps.py ..........                      [ 81%]
kaggle_environments/envs/tictactoe/test_tictactoe.py ...............     [100%]
```

File: kaggle_environments/envs/llm_20_questions/llm_20_questions.py [hunk 1]
```diff
@@ -171,42 +174,47 @@ def interpreter(state, env):
...
+        increment_turn(active1, inactive1, step, one_guessed)
+    
+    if active2.status in (TIMEOUT, ERROR) or two_bad_response:
+        end_game(active2, -1, active2.status)
+        end_game(inactive2, 1, DONE)
+    elif end_early or one_bad_response:
+        end_game(active2, 1, DONE)
+        end_game(inactive2, 1, DONE)
+    else:
+        increment_turn(active2, inactive2, step, two_guessed)
```
Comment (dster2) [inline kaggle_environments/envs/llm_20_questions/llm_20_questions.py:209]: Decent bit of mostly duplicated code that we could maybe unify in helper(s)?

**Review by dster2 — APPROVED** [review]
Thanks Bovard!

File: kaggle_environments/envs/llm_20_questions/llm_20_questions.py (hunk 1 above)
Comment (bovard) [reply kaggle_environments/envs/llm_20_questions/llm_20_questions.py:209]: yeah I've been meaning to do that, I'll do that in a follow up

**Pull request merged**